syntax = "proto3";
package caikit_data_model.nlp;

enum FinishReason {
  NOT_FINISHED = 0;
  MAX_TOKENS = 1;
  EOS_TOKEN = 2;
  CANCELLED = 3;
  TIME_LIMIT = 4;
  STOP_SEQUENCE = 5;
  TOKEN_LIMIT = 6;
  ERROR = 7;
}

enum InputWarningReason {
  UNSUITABLE_INPUT = 0;
}

message ChunkerTokenizationStreamResult {
  repeated caikit_data_model.nlp.Token results = 1;
  int64 token_count = 4;
  int64 processed_index = 2;
  int64 start_index = 3;
  int64 input_start_index = 20;
  int64 input_end_index = 21;
}

message Token {
  int64 start = 1;
  int64 end = 2;
  string text = 3;
}

message TokenizationResults {
  repeated caikit_data_model.nlp.Token results = 1;
  int64 token_count = 4;
}